experimental/devices/hx711/hx711.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package hx711 implements an interface to the HX711 analog to digital converter.
+//
+// Datasheet: https://www.mouser.com/ds/2/813/hx711_english-1022875.pdf
+package hx711
+
+import (
+	"errors"
+	"time"
+
+	"periph.io/x/periph/conn/gpio"
+)
+
+var (
+	// TimeoutError is returned from Read and ReadAveraged when the ADC took too
+	// long to indicate data was available.
+	TimeoutError = errors.New("timed out waiting for HX711 to become ready")
+)
+
+// InputMode controls the voltage gain and the channel multiplexer on the HX711.
+// Channel A can be used with a gain of 128 or 64, and Channel B can be used
+// with a gain of 32.
+//
+// Changing the InputMode on an HX711 will only take effect *after* the next
```

Comment:
Then maybe worth making it a method `SetGain()` so that the user doesn't have to care about this, as the function could do a wasted Read() implicitly. I feel it'd be simpler for users.